src/main/java/io/cdap/plugin/gcp/bigquery/sqlengine/BigQuerySQLEngine.java

```diff
@@ -169,11 +171,11 @@ public void prepareRun(SQLEngineContext context) throws Exception {
     configuration = BigQueryUtil.getBigQueryConfig(sqlEngineConfig.getServiceAccount(), sqlEngineConfig.getProject(),
                                                    cmekKeyName, sqlEngineConfig.getServiceAccountType());
     // Create resources needed for this execution
-    BigQuerySinkUtils.createResources(bigQuery, storage, DatasetId.of(datasetProject, dataset), bucket,
+    BigQuerySinkUtils.createResources(bigQuery, storage, DatasetId.of(datasetProject, datasetName), bucket,
                                       sqlEngineConfig.getLocation(), cmekKeyName);
     // Configure GCS bucket that is used to stage temporary files.
-    // If the bucket is created for this run, mar it for deletion after executon is completed
-    BigQuerySinkUtils.configureBucket(configuration, bucket, runId, sqlEngineConfig.getBucket() == null);
+    // If the bucket is created for this run, mark it for deletion after executon is completed
+    BigQuerySinkUtils.configureBucket(configuration, bucket, "bqpushdown-" + runId);
```

Comment:
nit: The second argument should be in a variable that outlines it's use

`String fallbackBucketName =  "bqpushdown-" + runId;`

Reply:
ack